Client/Classes/Recievers/ClientEventReciever.sqx

```diff
@@ -240,5 +240,31 @@ namespace Intrusion.Client
 				[_unit, assignedVehicleRole _unit select 0, _vehicle] call ClientSquadHandler.SetUniform;
 			};
 		};
+		
+		// Called when a box is attached to a vehicle.
+		// Adds a handler to destroy and delete it.
+		// The code in the handler is also in VehicleSpawner for the server.
+		// _vehicleObj (Object): The vehicle
+		// _attachObj (Object): The object attached to vehicle.
+		public static method OnVehicleBoxAttachmentCreated("_vehicleObj" as Object, "_attachObj" as Object)
+		{
+		    // When vehicle explodes, destroy ammo box
+			_vehicleObj addEventHandler ["Killed", {
```

Comment:
Too much code in a reciever method. The only code allowed here is group and unit filtering. All logic more than that should be passed into classes for each codes purpose. In this case there should be a new (static) method on the ClientEventHandler class. Maybe ClientEventHandler.AddEventHandlerThatRemovesWeaponBoxWhenVehicleKilled(...).

(This design pattern is not fully followed, but has been introduced lately. See the ServerEventHandler class to see the design pattern of methods that adds individual event handlers. The classes's purpose is to collect all event handler into the same class.)